### PR TITLE
feat: add contribution guide and CoC

### PR DESCRIPTION
**Pull request by kjappelbaum**

### CODE REVIEW

**Comment by kjappelbaum** [comment]
@MicPie it is still WIP, but you might see what I think might be a good setup. 

It is enough for "conventional" molecular->property datasets to know what the identifier and the property columns are. With description (and units), we can then build programmatically many different prompts. We might also try binning continuos data, etc. 

For data that is more "special" (e.g. reactions) we can still use the tag template you proposed

**Comment by kjappelbaum** [comment]
I also suggest a templating structure that I think is relatively flexible. 
@MicPie would be great if you could see if you also like this and if you can understand the explanations.

**Comment by kjappelbaum** [comment]
We also need to keep in mind that we need to have enough info to easily select the right tokenizer (https://github.com/OpenBioML/chemnlp/issues/9)

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+```yaml
+name: aquasoldb # unique identifier, we will also use this for directory names
+description: | # short description what this dataset is about
+  Curation of nine open source datasets on aqueous solubility.
+  The authors also assigned reliability groups.
+targets:
+  - id: Solubility # name of the column in a tabular dataset
+    description: Experimental aqueous solubility value (LogS) # description of what this column means
+    units: log(mol/L) # units of the values in this column (leave empty if unitless)
+    type: continuos # can be "categorical", "ordinal", "continuos"
```
Comment (MicPie) [inline CONTRIBUTING.md:47]: just a typo: continuos --> continuous

File: dev-requirements.tx [hunk 2]
```diff
@@ -0,0 +1,3 @@
+black>=22.1.0,<22.2.0
```
Comment (MicPie) [inline dev-requirements.tx:1]: This file has a `.tx` ending, should it be `.txt`?

**Review by MicPie — APPROVED** [review]

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+      - water solubility
+      - solubility in water
+identifiers:
+  - id: InChI # column name 
+    type: InChI # can be "SMILES", "SELFIES", "IUPAC", "OTHER"
+    description: International Chemical Identifier # description (optional, except for "OTHER")
+license: CC0 1.0 # license under which the original dataset was published
+links: # list of relevant links (original dataset, other uses, etc.)
+num_points: 10000 # number of datapoints in this dataset
+  - name: dataset
```
Comment (MicPie) [inline CONTRIBUTING.md:66]: Here the indentation is misaligned, or maybe a key is missing?

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+    url: https://dataverse.harvard.edu/dataset.xhtml?persistentId=doi:10.7910/DVN/OVHAW8
+    description: Original dataset
```
Comment (MicPie) [inline CONTRIBUTING.md:68]: above there is another description key

File: dev-requirements.tx (hunk 2 above)
Comment (kjappelbaum) [reply dev-requirements.tx:1]: thanks!

File: CONTRIBUTING.md (hunk 3 above)
Comment (kjappelbaum) [reply CONTRIBUTING.md:66]: will format the yaml, good point!

File: CONTRIBUTING.md (hunk 4 above)
Comment (kjappelbaum) [reply CONTRIBUTING.md:68]: The intention was that this description describes what the link is about. Will add a comment

**Pull request merged**